src/operator_repo/checks/__init__.py

```diff
@@ -126,10 +127,18 @@ def run_check(
     Run a check against a resource yielding all the problems found
     """
     log.debug("Running %s check on %s", check.__name__, target)
-    for result in check(target):
-        result.check = check.__name__
-        result.origin = target
-        yield result
+    try:
+        for result in check(target):
+            result.check = check.__name__
+            result.origin = target
+            yield result
+    except Exception:  # pylint: disable=broad-except
+        log.exception("Error running %s check on %s", check.__name__, target)
+        yield Fail(
+            traceback.format_exc(),
+            check=check.__name__,
+            origin=target,
+        )
```

Comment:
`traceback.format_exc()` can be quite long depending on the exception we hit and I don't think it makes sense to expose that to the end user since it is caused by a bug on our side.

What about this instead?

```suggestion
    except Exception as ex:  # pylint: disable=broad-except
        log.exception("Error running %s check on %s", check.__name__, target)
        yield Fail(
            f"Internal error: {repr(ex)}",
            check=check.__name__,
            origin=target,
        )
```

Reply:
I originally tried both `str()` or `repr()` but none of them showed enough deail to be able see what went wrong. For example in case of KeyError it only showed there was some key error but it wasn't possible to figure out where and why.

Reply:
to give you idea how it looks like in real output: https://github.com/Allda/community-operators-pipeline-preprod/pull/26#issuecomment-2653934616

Reply:
We already have that information in the logs: why do you think it adds any value to the user who sees that blob of text?

Reply:
Why not have it in both places? We and cert ops are also PR consumers so this saves us time going through logs.

Reply:
Alright. Can you bump the package version?

Reply:
Done